design/ceph-config-updates.md

```diff
@@ -0,0 +1,235 @@
+- `--run-dir` exists for all daemons, but it is likewise more desirable to use the `/var/run/ceph`
+  path in containers. Additionally, this directory stores only unix domain sockets, and it does not
+  need to be persisted to the host. We propose to simply use the `/var/run/ceph` location in
+  containers for runtime storage of the data.
+
+### Additional configuration
+
+Additional configuration which Rook sets up initially should be done by setting values in Ceph's
+centrally-stored config. A large group of additional configurations can be configured at once via
+the Ceph command `ceph config assimilage-conf`. Care should be taken to make sure that Rook does not
+modify preexisting user-specified values.
+**TODO** how to determine if a user has specified a value or if it is at default state?
+
+Existing global configs configured here:
+- `mon allow pool delete = true`
+- `fatal signal handlers = false`
+  **TODO** here unless it is required by virtue of the container context as with `log file`
+- `log stderr prefix = "debug "` should be set for all daemons to differentiate logging from auditing
+- `debug ...` configs
+
+Removed configs:
+- `mon_max_pg_per_osd = 1000` is a dangerous setting and should be removed regardless of whether
+  this proposal is accepted
+- `log file = /dev/stderr` is set by default to keep with container standards and kept here if the
+  user needs to change this for debugging/testing
+- `mon cluster log file = /dev/stderr` for `log file` reasons above
+- `mon keyvaluedb = rocksdb` is not needed for Luminous+ clusters
+- `filestore_omap_backend = rocksdb` is not needed for Luminous+
+- `osd pg bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pgp bits = 11` set (if needed) using config override for testing or play clusters
+- `osd pool default size = 1` set using config override for testing or play clusters
+- `osd pool default min size = 1` set using config override for testing or play clusters
+- `osd pool default pg num = 100` set (if needed) using config override for testing or play clusters
+- `osd pool default pgp num = 100` set (if needed) using config override for testing or play clusters
+- `rbd_default_features = 3`
+  **TODO** what is this changing and why? Does this need set after all?
+
+### Changes to Ceph mons
+
+The current method of starting mons where `mon-a` has an initial member of `a`, `mon-b` initial
+members `a b`, `mon-c` initial members `a b c`, etc. Has worked so far but could result in a race
+condition. Mon cluster stability is important to Ceph, and it is critical for this PR that the mons'
+centrally-stored config is stable, so we here note that this behavior should be fixed such that the
+mon initial members are known before the first mon is bootstrapped to consider this proposal's work
+completed. Practically speaking, this will merely require the mon services to be started and have
+IP addresses before the mons are bootstrapped.
```

Comment:
In a much earlier iteration of rook (perhaps v0.3), the mons were started with this method. The services were created first, then all three mon pods were started at the same time with the full list of mons. I ran into some issues where either it wasn't reliable or it was just much slower to start than when adding the mons one at a time to the quorum. In any case, it was probably an implementation bug and it would be good to go back to this approach.